test/apps/iodemo/io_demo.cc

```diff
@@ -922,8 +935,12 @@ class DemoClient : public P2pDemoCommon {
     }
 
     void terminate_connection(UcxConnection *conn, const char *reason) {
+        size_t server_index = get_server_index(conn);
+        if (server_index >= _server_info.size()) {
+            return;
+        }
```

Comment:
how can this happen?
maybe pass "server_index" to terminate_connection instead of conn. and lookup conn->index only in "dispatch_connection_error" ?